File: 01-data-structures/array/Array.java
public class Array {
  public static void main(String[] args) {
    // 定义一个包含5个整数的数组
    int[] arr = { 1, 2, 3, 4, 5 };

    // 访问数组元素
    System.out.println("Element at index 2: " + arr[2]);

    // 修改数组元素
    arr[2] = 10;
    System.out.println("Modified element at index 2: " + arr[2]);

    // 遍历数组
    for (int i : arr) {
      System.out.print(i + " ");
    }
    System.out.println();
  }
}

/*
 * jarry@MacBook-Pro array % java -version
 * openjdk version "23" 2024-09-17
 * OpenJDK Runtime Environment (build 23+37-2369)
 * OpenJDK 64-Bit Server VM (build 23+37-2369, mixed mode, sharing)
 * jarry@MacBook-Pro array % javac Array.java
 * jarry@MacBook-Pro array % java Array
 * Element at index 2: 3
 * Modified element at index 2: 10
 * 1 2 10 4 5
 */

File: 01-data-structures/array/array.c
#include <stdio.h>

int main()
{
    // 定义一个包含5个整数的数组
    int arr[5] = {1, 2, 3, 4, 5};

    // 访问数组元素
    printf("Element at index 2: %d\n", arr[2]);

    // 修改数组元素
    arr[2] = 10;
    printf("Modified element at index 2: %d\n", arr[2]);

    // 遍历数组
    for (int i = 0; i < 5; i++)
    {
        printf("%d ", arr[i]);
    }
    printf("\n");

    return 0;
}

/*
jarry@MacBook-Pro array % gcc -v
Apple clang version 16.0.0 (clang-1600.0.26.6)
Target: arm64-apple-darwin24.1.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
jarry@MacBook-Pro array % gcc array.c
jarry@MacBook-Pro array % ./a.out
Element at index 2: 3
Modified element at index 2: 10
1 2 10 4 5
*/